packages/signers/src/lit-protocol/signer.ts

```diff
@@ -0,0 +1,200 @@
+import { PKPEthersWallet } from "@lit-protocol/pkp-ethers";
+import { LitAbility, LitActionResource } from "@lit-protocol/auth-helpers";
+import {
+  type AuthCallbackParams,
+  type AuthSig,
+  type SessionSigsMap,
+} from "@lit-protocol/types";
+import { LitNodeClient } from "@lit-protocol/lit-node-client";
+import { generateSessionKeyPair } from "@lit-protocol/crypto";
+
+import {
+  type SignTypedDataParams,
+  type SmartAccountAuthenticator,
+} from "@alchemy/aa-core";
+import type { TypedDataDomain } from "viem";
+
+import {
+  type LitAuthMethod,
+  type LitSessionSigsMap,
+  type LitAccountAuthenticatorParams,
+  type LITAuthenticateProps,
+} from "./types.js";
+
+const SIGNER_TYPE: string = "lit";
+
+/**
+ * Implementation of `SmartAccountAuthenticator` for lit protocol
+ * This class requies:
+ * `@lit-protocol/lit-node-client`
+ * `@lit-protocol/pkp-ethers`
+ */
+export class LitSigner<C extends LitAuthMethod | LitSessionSigsMap>
+  implements
+    SmartAccountAuthenticator<
+      LITAuthenticateProps<C>,
+      LitSessionSigsMap,
+      PKPEthersWallet | undefined
+    >
+{
+  inner: PKPEthersWallet | undefined;
```

Comment:
this term might have been confusing. I thought of inner as the SDK necessary to produce a Signer. So in your case, it'd be what you construct in the constructor: the LitNodeClient.

then you'd have an inner signer field you'd be able to create with it's help, like you have below `private _signer: PKPEthersWallet | undefined;`

Reply:
Ah i see, this makes sense. will update